oauth2client/service_account.py

```diff
@@ -61,31 +61,49 @@ def __init__(self, service_account_id, service_account_email, private_key_id,
   def _generate_assertion(self):
     """Generate the assertion that will be used in the request."""
 
+    now = int(time.time())
     header = {
         'alg': 'RS256',
         'typ': 'JWT',
         'kid': self._private_key_id
     }
-
-    now = int(time.time())
     payload = {
         'aud': self._token_uri,
-        'scope': self._scopes,
         'iat': now,
         'exp': now + _ServiceAccountCredentials.MAX_TOKEN_LIFETIME_SECS,
         'iss': self._service_account_email
     }
-    payload.update(self._kwargs)
-
-    first_segment = _urlsafe_b64encode(_json_encode(header))
-    second_segment = _urlsafe_b64encode(_json_encode(payload))
-    assertion_input = first_segment + b'.' + second_segment
-
-    # Sign the assertion.
-    rsa_bytes = rsa.pkcs1.sign(assertion_input, self._private_key, 'SHA-256')
-    signature = base64.urlsafe_b64encode(rsa_bytes).rstrip(b'=')
-
-    return assertion_input + b'.' + signature
+    if self._scopes:
+      # Use usual process to acquire token
+      payload.update({
+          'scope': self._scopes,
+      })
+      payload.update(self._kwargs)
+
+      first_segment = _urlsafe_b64encode(_json_encode(header))
+      second_segment = _urlsafe_b64encode(_json_encode(payload))
+      assertion_input = first_segment + b'.' + second_segment
+
+      # Sign the assertion.
+      rsa_bytes = rsa.pkcs1.sign(assertion_input, self._private_key, 'SHA-256')
+      signature = base64.urlsafe_b64encode(rsa_bytes).rstrip(b'=')
+
+      return assertion_input + b'.' + signature
+
+    else:
```

Comment:
Entirely arbitrary. Earlier I made an attempt to get a clear spec on what it was that needed to be done with respect to this effort, and got unclear responses from multiple sources. That previous attempt used a separate class, but when I was finally informed that the credentials ought to be transparently 'promoted', I scrapped the approach.